soursarkar1990/sap-automation
# Cluster resources control & Terraform formatting (#408)

* Use jinja

* Add fail condition

* add correct comparison

* show disk free in GB

* select first node

* query for inode_available

* select only first node

* use json query

* correct the query

* don't convert

* Add the disk space assert

* make assert jinja

* correct the calculation

* convert to GB

* Fix calculation

* set the deployer free disk space threshhold

* make free space a parameter

* Add a dependency on the disk attachment resource

* remove the defaults

* Update vm-webdisp.tf

* Add support for dynamiv /hana/backup location

* Debugging local install volume creation

* JINJA fixes

* Remove the if statement

* adding is defined

* Add better messaging

* If workload zone defined AFS use it

* AFS download update

* Call AFS always

* Add SCS diskspace check

* /hana/data/backup tests

* Take the first item from the list

* Show mnt size

* Use index

* show raw data

* simplify the size calculation

* show backup location

* show backup directory

* Use the correct server

* Correct the if statement

* Move INI generation into the install block

* Register BoM in all cases

* Run az login earlier in the pipeline

* Script updates

* Show disk space requirement

* Only check size of /usr/sap if it exists

* Check if usr_sap is defined

* Disk size calculation

* missing line feed

* Add support for providing ORACLE-ASM

* Oracle ASM ports

* Oracle - ASM support

* Add support for Oracle ASM

* Add ORACLE-ASM to playbooks

* Oracle ASM support

* change UIDs

* move the check

* calculate scs_virtual_hostname

* remove extra part from tag

* become for profile

* use become for profile

* remove duplicates

* remove duplicates

* Update parameters to accommodate JAVA installations

* set become

* invoke cluster commands as root

* include_tasks instead of import_tasks

* test ACSS registration

* ACSS variable definitions and echo

* Improve echo :-)

* indentation

* Add the missing call

* pipeline updates

* Enable SLES 15 SP4

* quote the secret when passing from pipeline and in set_secrets.sh

* Disable Private FW

* Registery Configuration - Ciphers and Protocols

* Add memory dump configuration

* Move domain join to roles-os

* Add new OS configurations

* troubleshooting BoM

* Change the failed_when condition

* remove duplicate code

* debugging

* Ansible Lint

* troubleshooting

* Linting

* Linting

* Removed the ---

* Moved the tags

* Debugging

* Check directory exists

* Create tmp directory

* Fixed the Jinja

* ensure directories exist

* task naming

* BoM tests

* Linting fixes

* Testing

* Testing

* BoM registration

* Update common variable

* Test

* Revert download Dir

* Test

* Test

* Test

* Test

* Test

* Test

* Updated the disks_config.yml

* Playbook update - fetch the secrets

* Linting

* pass the secrets

* Task name tweaks

* Changed the directory logic

* Indentation

* SA tweaks

* Ignore storage account download errors

* Further testing

* JAVA Paths for logs

* Only validate BoM platform for deployment

* setting storage account flag

* Add the sa flag

* Add platform validation

* Log file paths

* pass the value back to set fact

* set the facts - slight modifications

* align to block

* align to block

* duh! errors

* indentation

* Commenting out the JAVA ports

* Removed the platform check

* support redeployment

* Test

* add region code for wus3

* Test

* Test

* Test

* Test

* Additional instance_type's added

* add the missing curly braces

* SCS ERS installation logic tweaks

* Calculate the second port

* format

* formatting woes

* check SCS & ERS creation

* Add better validation

* append to group

* adding limits for app tier

* remove sudoers

* block ERS install if SCS fails

* Test

* Test

* Adding OracleLinux8

* set sa_enabled flag

* Add default for instance_type

* Add BoM support to Web App

* Adding the insance type

* introduce dependency on disk attachment for extension

* don't check /usr/sap on Windows

* Windows fixes

* add code, need to refactor later, for JAVA HA profile updates before PAS installation

* determine ensa1 for java

* add backupdba group

* Logging & windows

* dgdba_gid

* Added dgdba_gid

* PAS log updates

* Added kmdba, racdba

* Add instance type to DBLoad

* Count the db virtual host name for all cases

* observer fixes

* added backupdba

* remove the /PD

* add racdba

* Moved variable definition

* Tweaks for InstanceType

* Don't use the download variable

* Linting

* Testing extraction

* Update NW_MSSQL_v0001ms-dbload-inifile-param.j2

* Update NW_MSSQL_v0001ms.yaml

Update the product ID for the DBLoad, SUM version, add SAPCAR

* update to use ansible.utils and changes to mss dbload jinja

* fix templating error in db2 mounts created during java deployments

* Wait for the processes to end

* pick up the right tasks

* Cluster checks

* Find sapcontrol

* changes to BOM and validation

* First edit of the cluster validation play

* Change directory name

* LD_LIBRARY_PATH update

* Vault DNS

* updates to cluster validation

* follow symlinks for find

* templating fixes for isntance type

* cut option typo

* become root for cluster commands

* missing ';'

* failed_when

* failed_when conditions

* sleep for 60 second

* cleanup the provisioning.

* simplify the code

* Change the download directory

* Revert

* home directory

* Add a fix for the unbound variable

* download paths

* comment out serialization

* remove the unneeded no_logs

* Set the unbound variables

* Add support for creating fileshares in existing storage accounts

* Terraform fmt for all files

* Test

* Test

---------

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>
Co-authored-by: hdamecharla <hdamecharla@microsoft.com>
Co-authored-by: Robert Biro <Robert.Biro@microsoft.com>
Co-authored-by: Nadeen Noaman <95418928+nnoaman@users.noreply.github.com>
Co-authored-by: Morgan Deegan <morgan@unicorn.town>
Co-authored-by: hdamecharla <71097261+hdamecharla@users.noreply.github.com>

## Problem
<Please describe what problem this PR is trying to resolve>

## Solution
<Please elaborate the solution for the problem>

## Tests
<Please provide steps to test the PR>

## Notes
<Additional comments for the PR>
